core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1889,6 +1891,9 @@ private long loadFileMetadataAndJournal(LockedInodePath inodePath,
     String ufsOwner = ufs.getOwner(ufsUri.toString());
     String ufsGroup = ufs.getGroup(ufsUri.toString());
     short ufsPermission = ufs.getMode(ufsUri.toString());
+    if (resolution.getShared()) {
+      ufsPermission |= (ufsPermission & (short) 0700) >> 6;
```

Comment:
this logic should be converted into a utility method in the permission class and given a meaningful name ... it is not obvious what is the purpose of this operation